### PR TITLE
JAMES-3261 ZIP packaging for all Guice servers

### PR DESCRIPTION
**Pull request by chibenwa**
**Why**: Allows to put the ZIPs on the download page (From next release! So likely 3.6.1... Once the backport is done...)

**Expected outcome**: More people should be using Guice products as a full compilation is no longer needed. This might be the first step to deprecate and eventually remove Spring packaging.

I had a productive Saturday:

 - Created a ZIP packaging that reuses the configuration files of the docker
 - Provided instructions to run the binary once unziped
 - Rewrote the docker images to rely on the ZIP - waiting for having JIB directly doing the work!
 - Manually tested everything...
 - Dropped the LDAP specialized docker image as their non-LDAP counter part now allows LDAP to be configured.
 
 Once released and available for uploads, existing install guides needs to be updated to rely on the uploads... (Not for today.)
 
 Enjoy!

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Review by jeantil — COMMENTED** [review]
Would it make sense to centralize the sample configuration in a reference/common directory ?
then only override the required files for each assembly ? 
I find it confusing to have multiple sample configurations not to mention the risk of drift if one file is edited in one place but not in the others ( I think there are already accidental and subtle differences between the various assemblies, on the mailetcontainer.xml for instance)

File: dockerfiles/run/guice/memory/Dockerfile [hunk 1]
```diff
@@ -22,7 +22,10 @@ RUN apt-get update && apt-get install -y unzip
 
 # Get data we need to run James : build results and configuration
 ADD destination/james-server-memory-guice.zip /root
-RUN unzip james-server-memory-guice.zip && cp -r /root/james-server-memory-guice/* /root
+RUN unzip james-server-memory-guice.zip && \
+  cp -r /root/james-server-memory-guice/* /root && \
+  rm james-server-memory-guice.zip && \
+  rm -r james-server-memory-guice
```
Comment (rouazana) [inline dockerfiles/run/guice/memory/Dockerfile:28]: unzip -d allows to extract to an arbitrary directory

Comment (chibenwa) [reply dockerfiles/run/guice/memory/Dockerfile:28]: Yes but it do not change the fact that everything is bundled in  a `james-server-memory-guice`

I tried:

```
RUN unzip james-server-memory-guice.zip -d /root && rm james-server-memory-guice.zip
```

It deflates:

```
Step 6/21 : RUN unzip james-server-memory-guice.zip -d /root && rm james-server-memory-guice.zip
 ---> Running in 4bfb050ef3e1
Archive:  james-server-memory-guice.zip
   creating: /root/james-server-memory-guice/
   creating: /root/james-server-memory-guice/extensions-jars/
   creating: /root/james-server-memory-guice/conf/
   creating: /root/james-server-memory-guice/james-server-memory-guice.lib/
  inflating: /root/james-server-memory-guice/README.adoc  
  inflating: /root/james-server-memory-guice/conf/jmx.properties  
  inflating: /root/james-server-memory-guice/conf/listeners.xml  
  inflating: /root/james-server-memory-guice/conf/logback.xml  
```

And I got the error:

```
$ docker run -ti mem
Error: Unable to access jarfile james-server-memory-guice.jar
```

Comment (rouazana) [reply dockerfiles/run/guice/memory/Dockerfile:28]: oh right sorry for the noise. It's easily doable with tar but unfortunately not with unzip.

**Comment by chibenwa** [comment]
It's open for weeks. The potentiality of conflicts high.

Committers, should I move forward and merge this?

**Comment by jeantil** [comment]
> Using symbolic links? If it works with the ZIP then I would be happy.

not really but this was simply a comment don't let it hold you

**Comment by Arsnael** [comment]
Needs a rebase

**Comment by chibenwa** [comment]
I just rebased to solve a trivial conflict.

If there is any additional comments on this, I would be happy to open follow up PRs.

**Pull request merged**